stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.xml

```diff
@@ -89,10 +89,56 @@
         >
             <attribute name="t-if">move_line.id</attribute>
         </xpath>
-        <!-- Remove "All items could not be shipped..." sentence -->
+        <!-- Remove "All items could not be shipped..." sentence
+            Substitute it by a message indicating if all goods have been delivered. -->
         <xpath expr='//div[hasclass("page")]/p[last()]' position="attributes">
             <attribute name="t-if">False</attribute>
         </xpath>
+        <xpath expr='//div[hasclass("page")]/p[last()]' position="after">
+            <t t-if="o.picking_type_id.code == 'outgoing'">
+                <p>Remaining to deliver:</p>
+                <t t-if="o.demand_is_completed()">
+                    <p>All the ordered items were delivered.</p>
+                </t>
+                <t t-else="">
+                    <table class="table table-sm" name="remaining_to_deliver_table">
+                        <tbody>
+                            <t
+                                t-set="pending_moves"
+                                t-value="get_remaining_to_deliver_data(o)"
+                            />
+                            <tr t-foreach="pending_moves" t-as="pending_move">
+                                <td>
+                                    <t
+                                        t-if="pending_move['is_header']"
+                                    ><![CDATA[<strong>]]></t>
```

Comment:
CDATA for?

Reply:
This is because of the IDE, that otherwise complains that the tag is not closed, and because of ```black```, that closes it (wrongly) when beautifying the code. If inside a CDATA, both problems are solved.

Reply:
Still. You don't need it. What's the final result you want?

Reply:
The tags <strong> and </strong> surround conditionally. If the move is a header one, then it's description is surrounded by <strong> and </strong>; otherwise, not. Maybe &lt; and &gt; work here. But I don't get what's wrong with CDATA.

Reply:
ok, then replace it with this:
```xml
<strong t-if="remaining['is_header']" t-esc="remaining['concept']" />
<span t-else="" t-esc="remaining['concept']" />
```
or even better if we don't care about the element type
```xml
<span t-att-class="'font-weight-bold' if remaining['is_header'] else None" t-esc="remaining['concept']" />
```

Reply:
I like very much your last suggestion. Thanks.